metron-platform/metron-indexing/src/main/java/org/apache/metron/indexing/dao/HBaseDao.java

```diff
@@ -138,9 +142,17 @@ private Document getDocumentFromResult(Result result) throws IOException {
     Map.Entry<byte[], byte[]> entry= columns.lastEntry();
     Long ts = Bytes.toLong(entry.getKey());
     if(entry.getValue()!= null) {
-      Map<String, Object> json = JSONUtils.INSTANCE.load(new String(entry.getValue()), new TypeReference<Map<String, Object>>() {
-      });
-      return new Document(json, Bytes.toString(result.getRow()), (String) json.get(SOURCE_TYPE), ts);
+      Map<String, Object> json = JSONUtils.INSTANCE.load(new String(entry.getValue()),
+          new TypeReference<Map<String, Object>>() {});
+      ByteArrayInputStream baos = new ByteArrayInputStream(result.getRow());
```

Comment:
Can we pull this out into a `readRowKey()` method, like we have a `buildRowKey()`?

Reply:
I initially tried to do this but if felt sort of awkward since we need to return both the guid and sensorType.  Would it make sense to just return a Document with those fields set and set the other fields after that method is called?  Or we could accept the json and ts as inputs to that method too.

Reply:
Ok, I submitted a [PR](https://github.com/merrimanr/incubator-metron/pull/34) against your branch that abstracts it out the key into an object and adds some explicit unit tests around it.